clash-ghc/src-ghc/Clash/GHC/PartialEval.hs

```diff
@@ -0,0 +1,111 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Clash.GHC.PartialEval
+  ( ghcEvaluator
+  ) where
+
+import Control.Monad.State.Strict (State)
+import qualified Control.Monad.State.Strict as State
+import GHC.Integer.GMP.Internals (Integer(..))
+
+import Clash.Core.DataCon
+import Clash.Core.Evaluator.Models
+import Clash.Core.Evaluator.Semantics
+import Clash.Core.Literal
+import Clash.Core.Term
+import Clash.Core.Type
+
+ghcEvaluator :: Evaluator
+ghcEvaluator = Evaluator ghcEvaluate ghcQuote
+
+ghcEvaluate :: Term -> State Env Value
```

Comment:
Add top-level comment. Say something about the architecture; i.e. why `ghcEvaluate` is passed to all of those other `evaluateXYZ` functions.